### ISSUE TITLE
Create data toolbar component

### ISSUE THREAD

**Issue by SeamusLeonardHPE** [grommet/hpe-design-system#3650]
Mapped to 2024 Q2 Objective 2: 
[Ship “toolbar” component](https://hpe.sharepoint.com/:p:/r/teams/hpe-design/_layouts/15/Doc.aspx?sourcedoc=%7B578BCD68-03C8-4054-88BC-B24EADABBE7A%7D&file=HPExs%20Goals%20Jan%202024%20-%20DS.pptx&action=edit&mobileredirect=true
)

[Building on work done by Vicky.](https://hpe.sharepoint.com/:p:/r/teams/hpe-design/_layouts/15/Doc.aspx?sourcedoc=%7B578BCD68-03C8-4054-88BC-B24EADABBE7A%7D&file=HPExs%20Goals%20Jan%202024%20-%20DS.pptx&action=edit&mobileredirect=true)

- [x] Check status of existing coded prototype with devs
- [x] Create anatomy of of toolbar component
- [x] Modify existing figma designs inc anatomy to match functionality or modification on coded prototype
- [ ] Note any dependencies (toggle view button, custom & saved view patterns)

Related tickets: 
[Data collection toolbar - Create coded prototype #3493](https://github.com/grommet/hpe-design-system/issues/3493)

[Map - Toggle view type button group in data collection toolbar #3432](https://github.com/grommet/hpe-design-system/issues/3432)

CodeSandbox: https://codesandbox.io/s/data-collection-toolbar-breakpoints-v1-b-ztzk7s
Deploy: https://ztzk7s.csb.app/

[WIP file ](https://www.figma.com/file/BsQor4xyGaU26jCmlXiZmw/Create-data-toolbar-component-%233650?type=design&node-id=0%3A1&mode=design&t=qBhVyJ64yVC8wNKZ-1)

Related ticket on data summary https://github.com/grommet/hpe-design-system/issues/3152

**Comment by ashifalinadaf** [comment]
Can you @taysea confirm the latest coded prototype that has been agreed upon 
is this the one? 
As mentioned in https://github.com/grommet/hpe-design-system/issues/3493 deliverables. 
'Coded prototype: Here is the prototype (v1.b from the below comment) that was most highly voted amongst the team:
CodeSandbox: https://codesandbox.io/s/data-collection-toolbar-breakpoints-v1-b-ztzk7s
Deploy: https://ztzk7s.csb.app/'

**Comment by ashifalinadaf** [comment]
[WIP file ](https://www.figma.com/file/BsQor4xyGaU26jCmlXiZmw/Create-data-toolbar-component-%233650?type=design&node-id=0%3A1&mode=design&t=9Y5bYtj6yH3lPDLb-1)

**Comment by taysea** [comment]
> @taysea can you please confirm the latest coded prototype that has been agreed upon is this the one? As mentioned in #3493 deliverables.
> 
> > 'Coded prototype: Here is the prototype (v1.b from the below comment) that was most highly voted amongst the team:
> > CodeSandbox: https://codesandbox.io/s/data-collection-toolbar-breakpoints-v1-b-ztzk7s
> > Deploy: https://ztzk7s.csb.app/'

Hi Ashif, this is the correct link. The theme styling is using the old styling, so use the latest from v5.1 as you design.

**Comment by SeamusLeonardHPE** [comment]
@ashifalinadaf since we have paused work on "saved view"

Can you please look at the data tool bar component again:

- Define what customisations are available to each view mode (eg does "Sort" button disappear when in map or chart view mode). This can be done in bullet list.
- Review current DS guidance page (figma scrape linked below) and suggest edits to bring guidance into alignment with updated design 

[https://www.figma.com/file/H47lk4zrHgkkRvwV9EIzq4/%233650---Create-data-toolbar-component?type=design&node-
id=0%3A1&mode=design&t=K8j9Jtxdk0SIkvuK-1](https://www.figma.com/file/uZD8NwEmL1cdt3SEbT5Bt6/%233650---Create-Data-Collection-Toolbar?type=design&node-id=2661-13124&mode=design)

**Comment by SeamusLeonardHPE** [comment]
Update for team.

2 versions of the toolbar have been shared amongst the product design team and we are awaiting feedback to their preferred usage.

https://www.figma.com/file/AGC2ntsUG3hEU7QFxD8tkZ/%233650---DataToolbar-Demo?type=design&node-id=0%3A1&mode=design&t=CqfWyGnAnXauc9OP-1

In version 1 we created sub components with all the logic for state & breakpoint & allow the designer to access these form the parent component. ("Expose properties from Nested instances" is how this is described int the parent component property)

In version 2 we provided subcomponents, but the designer needs to be select inside the components to access the nested variants

![Image](https://github.com/grommet/hpe-design-system/assets/135040188/ab9b3c86-174e-4da8-a6cb-2a100c264f0a)
